File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/dao/repository/EducatorReadRepository.java
package io.say.better.storage.mysql.dao.repository;

import io.say.better.core.enums.Provider;
import io.say.better.storage.mysql.domain.entity.Educator;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface EducatorReadRepository extends JpaRepository<Educator, Long>, MemberReadRepository<Educator> {

	Optional<Educator> findByProviderAndLoginId(Provider provider, String loginId);

	Optional<Educator> findByEmail(String email);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/dao/repository/EducatorReadRepository.kt
package io.say.better.storage.mysql.dao.repository

import io.say.better.storage.mysql.domain.entity.Educator
import io.say.better.storage.mysql.domain.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface EducatorReadRepository : JpaRepository<Educator, Long> {
    fun findByEducatorId(educatorId: Long): Optional<Educator>

    fun findByMemberId(member: Member): Optional<Educator>
}
